core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -174,18 +242,39 @@ private[impl] trait DataCutterParams extends Params {
   private[op] final val labelsToKeep = new DoubleArrayParam(this, "labelsToKeep",
     "labels to keep when applying the data cutter")
 
-  private[op] def setLabels(keep: Set[Double], drop: Set[Double]): this.type = {
-    set(labelsToKeep, keep.toArray.sorted)
-    set(labelsToDrop, drop.toArray.sorted)
+  private[op] def setLabels(keep: Seq[Double], dropTopK: Seq[Double], labelsDropped: Long): this.type = {
+    set(labelsToKeep, keep.toArray)
+      .set(labelsToDrop, dropTopK.toArray)
+      .set(labelsDroppedTotal, labelsDropped)
   }
 
   private[op] def getLabelsToKeep: Array[Double] = $(labelsToKeep)
 
   private[op] final val labelsToDrop = new DoubleArrayParam(this, "labelsDropped",
-    "labels to drop when applying the data cutter")
+    "the top of the labels to drop when applying the data cutter")
 
   private[op] def getLabelsToDrop: Array[Double] = $(labelsToDrop)
 
+  private[op] final val labelsDroppedTotal = new LongParam(this, "labelsDroppedTotal",
+    "the number of labels dropped")
+
+  private[op] def getLabelsDroppedTotal: Long = $(labelsDroppedTotal)
+
+  final val maxLabelsDroppedForDiagnostics = new IntParam(this, "maxLabelsDroppedForDiagnostics",
+    "maximum number of dropped label categories to retain for logging",
+    ParamValidators.inRange(lowerBound = 0, upperBound = 100, lowerInclusive = true, upperInclusive = true)
```

Comment:
is there a reason why do we set 100 as an upper limit?

Reply:
This is the number of names of dropped labels that you keep? can we change the name to reflect that? maybe maxNamesForDroppedLabels

Reply:
@tovbinm it's a reasonable limit to log. Int.MaxValue is unreasonable

Reply:
Look artificial to me and should perhaps be relative to data size. let's extend it to 1000 at least.

Reply:
anything proportional to data size will have the effect of creating huge un-GC'able  collections in degenerate cases, thus a hard cap on top of it is also necessary. Less of a concern if we just pipe it to logging instead of storing it in a param.